### ISSUE TITLE
timesyncd:  there is no way to differentiate NTP servers per-interface when multiple interfaces configured

### ISSUE THREAD

**Issue by raviteja-b** [systemd/systemd#33224]
### systemd version the issue has been seen with

255

### Used distribution

 GNU/Linux

### Linux kernel version used

6.6.30-a62f665-ga62f66583c58

### CPU architectures issue was seen on

arm

### Component

systemd

### Expected behaviour you didn't see

When there are multiple ethernet interfaces configured with systemd-networkd and one ethernet interface has got DHCP provided NTP servers then its important to have a way to differentiate which ethernet interface has got DHCP provided NTP servers on time-syncd.
but currently there is no way to identify which ethernet interface has got ntp servers on time-synod

### Unexpected behaviour you saw

currently there is no way to identify which ethernet interface has got ntp servers on time-syncd

### Steps to reproduce the problem

1. Configure multiple ethernet interface with systemd-networkd
2. Configure DHCP on one interface with NTPServers configured on DHCP servers.
3. noticed signal when timesyncd LinkNTPServers property get updated
    like described in this PR https://github.com/systemd/systemd/pull/27845
4. there is no way to identify to which ethernet interface these DHCP server provided ntp servers belongs to.

### Additional program output to the terminal or log subsystem illustrating the issue

```sh
D-bus signal looks like below example, but there is no way to identify ethernet interface

 MESSAGE "sa{sv}as" {
          STRING "org.freedesktop.timesync1.Manager";
          ARRAY "{sv}" {
                  DICT_ENTRY "sv" {
                          STRING "LinkNTPServers";
                          VARIANT "as" {
                                  ARRAY "s" {
                                          STRING "10.8.8.18";
                                          STRING "10.8.8.19";
                                  };
                          };
                  };
          };
          ARRAY "s" {
          };
```

**Comment by raviteja-b** [comment]
@yuwata can you share you views on this behaviour

**Comment by yuwata** [comment]
Yeah, like `resolvectl`, we should provide such information through `timedatectl timesync-status` or a new command.

**Comment by raviteja-b** [comment]
> Yeah, like `resolvectl`, we should provide such information through `timedatectl timesync-status` or a new command.

as part of D-bus event,  it should  have parameter to know to which ethernet interface these network provided  NTP servers belongs to.
